drivers/clock_control/clock_stm32_ll_h5.c

```diff
@@ -0,0 +1,769 @@
+					  clock_control_subsys_t sub_system)
+{
+	struct stm32_pclken *pclken = (struct stm32_pclken *)(sub_system);
+	volatile uint32_t *reg;
+	uint32_t reg_val;
+
+	ARG_UNUSED(dev);
+
+	if (IN_RANGE(pclken->bus, STM32_PERIPH_BUS_MIN, STM32_PERIPH_BUS_MAX) == 0) {
+		/* Attemp to toggle a wrong periph clock bit */
+		return -ENOTSUP;
+	}
+
+	reg = (uint32_t *)(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus);
+	reg_val = *reg;
+	reg_val &= ~pclken->enr;
+	*reg = reg_val;
+
+	return 0;
+}
+
+static inline int stm32_clock_control_configure(const struct device *dev,
+						clock_control_subsys_t sub_system,
+						void *data)
+{
+	struct stm32_pclken *pclken = (struct stm32_pclken *)(sub_system);
+	volatile uint32_t *reg;
+	uint32_t reg_val, dt_val;
+	int err;
+
+	ARG_UNUSED(dev);
+	ARG_UNUSED(data);
+
+	err = enabled_clock(pclken->bus);
+	if (err < 0) {
+		/* Attempt to configure a src clock not available or not valid */
+		return err;
+	}
+
+	dt_val = STM32_CLOCK_VAL_GET(pclken->enr) <<
+					STM32_CLOCK_SHIFT_GET(pclken->enr);
+	reg = (uint32_t *)(DT_REG_ADDR(DT_NODELABEL(rcc)) +
+					STM32_CLOCK_REG_GET(pclken->enr));
+	reg_val = *reg;
+	reg_val |= dt_val;
+	*reg = reg_val;
+
+	return 0;
+}
+
+static int stm32_clock_control_get_subsys_rate(const struct device *dev,
+					       clock_control_subsys_t sys,
+					       uint32_t *rate)
+{
+	struct stm32_pclken *pclken = (struct stm32_pclken *)(sys);
+
+	/*
+	 * Get AHB Clock (= SystemCoreClock = SYSCLK/prescaler)
+	 * SystemCoreClock is preferred to CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
+	 * since it will be updated after clock configuration and hence
+	 * more likely to contain actual clock speed
+	 */
+	uint32_t ahb_clock = SystemCoreClock;
+	uint32_t apb1_clock = get_bus_clock(ahb_clock, STM32_APB1_PRESCALER);
+	uint32_t apb2_clock = get_bus_clock(ahb_clock, STM32_APB2_PRESCALER);
+	uint32_t apb3_clock = get_bus_clock(ahb_clock, STM32_APB3_PRESCALER);
+
+	ARG_UNUSED(dev);
+
+	switch (pclken->bus) {
+	case STM32_CLOCK_BUS_AHB1:
+	case STM32_CLOCK_BUS_AHB2:
+	case STM32_CLOCK_BUS_AHB4:
+		*rate = ahb_clock;
+		break;
+	case STM32_CLOCK_BUS_APB1:
+	case STM32_CLOCK_BUS_APB1_2:
+		*rate = apb1_clock;
+		break;
+	case STM32_CLOCK_BUS_APB2:
+		*rate = apb2_clock;
+		break;
+	case STM32_CLOCK_BUS_APB3:
+		*rate = apb3_clock;
+		break;
+	case STM32_SRC_SYSCLK:
+		*rate = get_sysclk_frequency();
+		break;
+#if defined(STM32_HSI_ENABLED)
+	case STM32_SRC_HSI:
+		*rate = STM32_HSI_FREQ;
+		break;
+#endif /* STM32_HSI_ENABLED */
+#if defined(STM32_CSI_ENABLED)
+	case STM32_SRC_CSI:
+		*rate = STM32_CSI_FREQ;
+		break;
+#endif /* STM32_MSIS_ENABLED */
+#if defined(STM32_HSE_ENABLED)
+	case STM32_SRC_HSE:
+		*rate = STM32_HSE_FREQ;
+		break;
+#endif /* STM32_HSE_ENABLED */
+#if defined(STM32_LSE_ENABLED)
+	case STM32_SRC_LSE:
+		*rate = STM32_LSE_FREQ;
+		break;
+#endif /* STM32_LSE_ENABLED */
+#if defined(STM32_LSI_ENABLED)
+	case STM32_SRC_LSI:
+		*rate = STM32_LSI_FREQ;
+		break;
+#endif /* STM32_LSI_ENABLED */
```

Comment:
non blocking, but in case you are addressing Erwan's comment you could add the following HSI48 snipped in the same go. 
```suggestion
#endif /* STM32_LSI_ENABLED */
#if defined(STM32_HSI48_ENABLED)
	case STM32_SRC_HSI48:
		*rate = STM32_HSI48_FREQ;
		break;
#endif /* STM32_HSI48_ENABLED */
```

Reply:
done